src/passes/SimplifyLocals.cpp

```diff
@@ -754,13 +760,14 @@ struct SimplifyLocals
     *item = builder.makeNop();
     ifTrueBlock->finalize();
     assert(ifTrueBlock->type != Type::none);
+    auto localType = this->getFunction()->getLocalType(set->index);
     // Update the ifFalse side.
-    iff->ifFalse = builder.makeLocalGet(
-      set->index, this->getFunction()->getLocalType(set->index));
-    iff->finalize(); // update type
+    iff->ifFalse = builder.makeLocalGet(set->index, localType);
+    // Update the type of the if to that of the local.
+    iff->finalize(localType);
```

Comment:
Is this one always precise? If not, it would be good to add a TODO like on the previous one.